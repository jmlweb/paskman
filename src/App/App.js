// @flow
import * as React from 'react';
import styled, { injectGlobal } from 'styled-components';
import { normalize } from 'polished';
import Resizer from './Resizer/ResizerContainer';
import TopBar from './TopBar/TopBarContainer';
import Router from './Router/Router';
import routes from '../routes';
import fonts, { fontSmoothing } from '../styles/fonts';

const { rootFontSize, defaultFontSize, fontFamily } = fonts;

injectGlobal`${normalize()}`;

injectGlobal`
html {
  font-size: ${rootFontSize};
}

body {
  font-family: ${fontFamily};
  ${fontSmoothing()}
}
`;

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  font-size: ${defaultFontSize};
  height: 100vh;
  width: 100%;
`;

const App = () => (
  <Wrapper>
    <Resizer />
    <TopBar routes={routes} />
    <Router routes={routes} />
  </Wrapper>
);

export default App;
