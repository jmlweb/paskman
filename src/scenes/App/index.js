// @flow
import React from 'react';
import { Route, Link } from 'react-router-dom';
import createDynamicComponent from '../../components/DynamicComponent';

type RouteType = {
  +path: string,
  +title: string,
  +component: any,
};

const routes: Array<RouteType> = [
  {
    path: '/',
    title: 'Home',
    component: createDynamicComponent(
      () => import('../Home'), () => require('../Home')
    ),
  },
];

const App = () => (
  <div>
    <header>
      {routes.map(route => <Link key={`link${route.title}`} to={route.path}>{route.title}</Link>)}
    </header>

    <main>
      {routes.map(route => <Route key={`route${route.title}`} exact path={route.path} component={route.component} />)}
    </main>
  </div>
);

export default App;
