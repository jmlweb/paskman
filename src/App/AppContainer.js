import React, { PureComponent } from 'react';
import PT from 'prop-types';
import { connect } from 'react-redux';
import { settingsFetch } from '../data/settings/duck';
import routes from '../routes';
import App from './App';

class AppContainer extends PureComponent {
  static propTypes = {
    settingsFetch: PT.func.isRequired,
  }
  componentDidMount() {
    this.props.settingsFetch();
  }
  render() {
    return <App routes={routes} />;
  }
}

export const mapStateToProps = state => ({
  location: state.routing.location,
  settings: state.data.settings,
});

export const mapDispatchToProps = {
  settingsFetch,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(AppContainer);
