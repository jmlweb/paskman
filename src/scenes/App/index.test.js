import React from 'react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'react-router-redux';
import createHistory from 'history/createBrowserHistory';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import App from './';

configure({ adapter: new Adapter() });

describe('App', () => {
  const mockStore = configureStore();
  let store;
  let history;
  let wrapper;
  let count = 0;
  beforeEach(() => {
    store = mockStore({});
    history = createHistory();
    wrapper = mount(
      <Provider key={count} store={store}>
        <ConnectedRouter key={count} history={history}>
          <App key={count} />
        </ConnectedRouter>
      </Provider>
    );
    count += 1;
  });
  it('Render the connected component', () => {
    expect(wrapper.find(App).length).toEqual(1);
  });
  it('Matchs snapshot', () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
