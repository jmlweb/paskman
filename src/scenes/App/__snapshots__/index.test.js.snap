// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Matchs snapshot 1`] = `
<div>
  <header>
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </header>
  <main />
</div>
`;
